GPy/kern/src/symmetric.py

```diff
@@ -0,0 +1,137 @@
+import numpy as np
+
+from .kern import Kern
+
+
+class Symmetric(Kern):
+    """
+    Symmetric kernel that models a function with even or odd symmetry:
+
+    .. math::
+
+        f(x) = f(Ax)
+
+    or
+
+    .. math::
+
+        f(x) = -f(Ax)
+
+    it does this by modelling:
+
+    .. math::
+
+        f(x) = g(x) \pm g(Ax)
+
+    with kernel
+
+    .. math::
+
+        k(x, x') \pm k(Ax, x') \pm k(x, Ax') + k(Ax, Ax')
+
+    where k(x, x') is the kernel of g(x)
+    """
```

Comment:
I can get the definition of this kernel, but for a person who is less familiar with GP it would be confusing. Please expand the description with a bit more details, e.g. "\pm" is confusing, it is better to use the sign variable or something.